### DISCUSSION TITLE
TypeError: 'type' object is not subscriptable

### DISCUSSION THREAD

**Discussion by Rad-007** [smartcontractkit/full-blockchain-solidity-course-py#1333] [Q&A] [answered]
```
PS C:\Users\dell\OneDrive\Project\lottery> brownie test --network mainnet-fork
INFO: Could not find files for the given pattern(s).
Brownie v1.18.1 - Python development framework for Ethereum

========================================================== test session starts ===========================================================
platform win32 -- Python 3.9.6, pytest-6.2.5, py-1.11.0, pluggy-1.0.0
rootdir: C:\Users\dell\OneDrive\Project\lottery
================================================================ FAILURES ================================================================ 
__________________________________________________________ test_get_entrancefee __________________________________________________________ 

    def test_get_entrancefee():        account=accounts[0]
>       lottery=Lottery.deploy(config["networks"][network.show_active()]["eth_usd_price_feed"],{"from":account}

        )
E       TypeError: 'type' object is not subscriptable

tests\test_lottery.py:11: TypeError
======================================================== short test summary info ========================================================= 
FAILED tests/test_lottery.py::test_get_entrancefee - TypeError: 'type' object is not subscriptable
=========================================================== 1 failed in 11.42s =========================================================== 
Terminating local RPC client...

```

**Comment by cromewar** [comment]
Hello @Rad-007 
We will need more info on this, please review #980 and reformulate your question.

**Reply by Rad-007** [reply]
```
 def test_get_entrancefee():        account=accounts[0]
>       lottery=Lottery.deploy(config["networks"][network.show_active()]["eth_usd_price_feed"],{"from":account}

        )
E       TypeError: 'type' object is not subscriptable

tests\test_lottery.py:11: TypeError
======================================================== short test summary info ========================================================= 
FAILED tests/test_lottery.py::test_get_entrancefee - TypeError: 'type' object is not subscriptable
=========================================================== 1 failed in 11.42s =========================================================== 
Terminating local RPC client...

```

Here I am gerrting error 
Could you please tell me 
<img width="786" alt="Screenshot 2022-03-24 130827" src="https://user-images.githubusercontent.com/84834907/160062604-cb657183-2c2c-460b-8520-987e488388b6.png">
what is wrong here and how to fix it

**Reply by cromewar** [reply]
The error is on this line:
`lottery=Lottery.deploy(config["networks"][network.show_active()]["eth_usd_price_feed"]`
Check your `brownie-config.yaml` and see if you have the `eth_usd_price_feed` for the network you are trying to deploy to.

**Reply by Rad-007** [reply]
Yes I have
eth_usd_price_feed in brownie-config.yaml

**Comment by Rad-007** [comment]
<img width="532" alt="image" src="https://user-images.githubusercontent.com/84834907/160225765-a8a7ef0d-38a0-435f-9d8a-b3fba68e3576.png">


I am getting 2 errors could you please tell

**Reply by cromewar** [reply]
You must add `0x` add the beggining of the eth address otherwise python is not going to recognize it.

**Comment by Rad-007** [comment]
It's already there
<img width="483" alt="image" src="https://user-images.githubusercontent.com/84834907/160266479-e1ffbd9c-b15f-4ff0-a0ec-23de0cbbc72e.png">

**Comment by cromewar** [comment]
Please share the entire deploy script, and also the contract code.. The error is pointing to the constructor which I'm almost sure is due to a typo or bad syntax but I'll need to check your code in order to corroborate.

**Comment by Rad-007** [comment]
<img width="341" alt="image" src="https://user-images.githubusercontent.com/84834907/160537015-cc5f7af4-4fb1-41c8-9bbf-8c7b059836a4.png">



<img width="449" alt="image" src="https://user-images.githubusercontent.com/84834907/160537058-77f86747-801d-4847-8337-84c3e4eea1e5.png">

**Comment by cromewar** [comment, marked as answer]
After looking into for code I know what the actual problem is.
On your brownie config you have the `eth_usd_price_feed` without quotes you must add them `''` please check the code and compare with yours:
https://github.com/PatrickAlphaC/brownie_fund_me/blob/main/brownie-config.yaml

**Comment by Rad-007** [comment]
I am getting a new error


![image](https://user-images.githubusercontent.com/84834907/161491275-b7751e95-9f45-4b67-ab7e-85c8789893f9.png)

and what is wrong here


![image](https://user-images.githubusercontent.com/84834907/161491371-962dd32d-5db5-4f8d-840e-2241bca9b2b3.png)
![image](https://user-images.githubusercontent.com/84834907/161491406-d49b092a-e072-4eeb-b7fb-2c1fc3460c96.png)

**Reply by cromewar** [reply]
Seems you have an extra/non well described import on your python file, something like this:
![image](https://user-images.githubusercontent.com/20199180/161589588-b38b81f9-7bde-4e62-add9-1eb26d89f26b.png)
and on your env file, remember to name it `.env` also on windows the export statement is not necessary.

**Comment by Rad-007** [comment]
![image](https://user-images.githubusercontent.com/84834907/162287488-56661fd5-8d06-4b30-909d-986034e33c39.png)


What is wrong here 
Could u please tell

**Reply by cromewar** [reply]
Rename the contract to LinkToken.sol and also be sure this is the same name as on the contract... If you import Link_Token and the contract name (in the code) is LInkToken, you will not be able to import it.

**Comment by Rad-007** [comment]
![image](https://user-images.githubusercontent.com/84834907/162495651-d805d739-6d4b-42a3-b998-7ca6cac15e29.png)


here is my get account function

![image](https://user-images.githubusercontent.com/84834907/162495796-7599404f-ca8c-4398-8524-27ccd9184f5e.png)

**Comment by Rad-007** [comment]
I forgot the password of an account 
How can I reset or change it

**Reply by cromewar** [reply]
Hello @Rad-007 
Just delete the network and add it again:
![image](https://user-images.githubusercontent.com/20199180/165002317-f83d75db-f4dd-4c5b-a8a2-6d5e304abdbb.png)

**Comment by Rad-007** [comment]
I have some issues related to openzepplin

![image](https://user-images.githubusercontent.com/84834907/168036374-34658d2e-c515-4f9b-a3a7-aee09d04c137.png)

Help me

**Reply by cromewar** [reply]
Hello @Rad-007 
the version of openzeppelin is 3.4.0 no 3.0.0 
And also remember to add the SPDX license identifier on the contracts ;)

**Comment by Rad-007** [comment]
![image](https://user-images.githubusercontent.com/84834907/168758958-da5b7a07-9c89-4136-b6d0-47032a90e678.png)


Any idea why?

**Reply by cromewar** [reply]
this happens type to time with brownie not being able to select the correct version I strongly suggest to delete the `.brownie` file on your home path, on your case `C:\Users\dell`.... this is a hidden folder to don't forget to show the hidden files on windows explorer.
Cheers.

**Comment by Rad-007** [comment]
<img width="816" alt="image" src="https://user-images.githubusercontent.com/84834907/171988165-b24e7448-df66-4d5b-9de0-5cce02be7e46.png">


<img width="786" alt="image" src="https://user-images.githubusercontent.com/84834907/171988187-2fc0f23a-36a0-4c56-a0f7-17ac89dcb53c.png">

**Reply by Rad-007** [reply]
Help, please!
I have been stuck for while

**Reply by cromewar** [reply]
This is the cytools error, pretty common please refer to chronological issues and look up "python related problems".
